kytos/core/id.py

```diff
@@ -0,0 +1,29 @@
+class SwitchID(str):
+    pass
+
+class InterfaceID(str):
+    __slots__ = ("switch", "port")
+    def __new__(cls, switch:SwitchID, port:int):
+        return super().__new__(cls, f"{switch}:{port}")
+    def __init__(self, switch:SwitchID, port:int):
+        #Used for sorting, but can be accessed
+        self.switch = switch
+        self.port = port
+
+    def __lt__(self, other):
+        # Ensures that instances are sortable in a way that maintains backwards
+        # compatibility when creating instances of LinkID
+        dpid_a = self.switch
+        port_a = self.port
+        dpid_b = other.switch
+        port_b = other.port
+        if dpid_a < dpid_b:
+            return True
+        elif dpid_a == dpid_b and port_a < port_b:
+            return True
+        else:
+            return False
+
+class LinkID(str):
+    def __new__(cls, interface_1:InterfaceID, interface_2:InterfaceID):
+        return super().__new__(cls, ":".join(sorted((interface_1, interface_2))))
```

Comment:
@Ktmi, as it is it would break compatibility, Link id used o be a sha256 string, could you also include calling hashlib.sha256 here and getting the hexdigest?

Reply:
Sure, I can do that. But I do have a question, why do we use the sha256 hash of the string over the string itself? Is this a security requirement?

Reply:
Nope, it was not driven by security. Originally, it used to be an uuid4, but then for facilitating equality comparison reasons it evolved to be deterministic based on endpoints, sha256 was chosen to meet this requirement.